### PR TITLE
To fix IDETECT-4023 : Change purgeUnmatchedFilesEnabled field type fr…

### PR DESCRIPTION
**Pull request by gopannabd**
Change purgeUnmatchedFilesEnabled field type from Varies to String for ProjectView API.
Resolve IDETECT-4023.

### CODE REVIEW

File: src/main/java/com/synopsys/integration/blackduck/api/generated/discovery/ApiDiscovery.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
 import com.synopsys.integration.blackduck.api.generated.view.OriginDependencyPathsView;
 import com.synopsys.integration.blackduck.api.generated.view.PolicyRuleView;
 import com.synopsys.integration.blackduck.api.generated.view.ProjectGroupsView;
-import com.synopsys.integration.blackduck.api.generated.view.ProjectView;
+import com.synopsys.integration.blackduck.api.manual.view.ProjectView;
```
Comment (gopannabd) [inline src/main/java/com/synopsys/integration/blackduck/api/generated/discovery/ApiDiscovery.java:40]: The ProjectView class is moved to Manual package from Generated while HUB specification remains inconsistent.

File: src/main/java/com/synopsys/integration/blackduck/api/manual/view/ProjectView.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public class ProjectView extends BlackDuckView {
     private Boolean projectLevelAdjustments;
     private String projectOwner;
     private Integer projectTier;
-    private Varies purgeUnmatchedFilesEnabled;
+    private String purgeUnmatchedFilesEnabled;
```
Comment (gopannabd) [inline src/main/java/com/synopsys/integration/blackduck/api/manual/view/ProjectView.java:68]: The change required to fix the API for ProjectView.

**Comment by ronaklotia-sys** [comment]
#LGTM 

but we should have someone from Alert Team too to review this too.

**Review by ronaklotia-sys — APPROVED** [review]

**Comment by ChomickiM** [comment]
Is it possible to keep the existing generated class but add a new class that is a duplicate with the fixes in the manual package? If this issue is in Blackduck and is scheduled to be fixed in the future, the generated class should eventually grab those changes. This way we still get any other updates to this endpoint without having to manually validate if changes were made with each release.

Alert does consume the ProjectView but not specifically the `purgeUnmatchedFilesEnabled` field. If we made the change this way then I believe Alert should be able to continue using the generated object. Detect would then just need to update to using the manual until the fix is made by the Blackduck API team.

**Comment by gopannabd** [comment]
They made the purgeUnmatchedFilesEnabled
Not nullable in 2023.4.2. It is broken in the sense that we get a string value in the actual JSON response but it is defined as a Varies in the generated ProjectView, even if we do not really use this field. This should be broken for Alert too (from BD HUB 2023.4.2).

Best,
Madhu
From: Martin ***@***.***>
Date: Friday, September 1, 2023 at 9:26 AM
To: blackducksoftware/blackduck-common-api ***@***.***>
Cc: Madhu Gopanna ***@***.***>, Assign ***@***.***>
Subject: Re: [blackducksoftware/blackduck-common-api] To fix IDETECT-4023 : Change purgeUnmatchedFilesEnabled field type fr… (PR #56)

Is it possible to keep the existing generated class but add a new class that is a duplicate with the fixes in the manual package? If this issue is in Blackduck and is scheduled to be fixed in the future, the generated class should eventually grab those changes. This way we still get any other updates to this endpoint without having to manually validate if changes were made with each release.

Alert does consume the ProjectView but not specifically the purgeUnmatchedFilesEnabled field. If we made the change this way then I believe Alert should be able to continue using the generated object. Detect would then just need to update to using the manual until the fix is made by the Blackduck API team.

**Comment by psantos1113** [comment]
No real issues with the actual code changes. You need to change blackduck-common with a corresponding PR to change all the references to ProjectView to the manual class.

**Comment by psantos1113** [comment]
Why can't you just update the api generator with a single line change to override the media type?

`mediaTypeOverrides.put("application/vnd.blackducksoftware.project-detail-6+json", "application/vnd.blackducksoftware.project-detail-5+json")`
Add that one line to MediaTypePathManager and it will use version 5 of the project details.  When Black Duck fixes the API you will need to run the generator to build an updated API you can remove that line when the issue is fixed.  Then none of the downstream projects need to change their codebase.

**Review by psantos1113 — APPROVED** [review]

**Pull request merged**